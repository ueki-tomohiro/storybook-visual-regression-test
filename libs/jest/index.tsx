import { composeStories, StoryFile } from "@storybook/testing-react";
import { render } from "@testing-library/react";
import React from "react";

export const renderFromStories = (stories: StoryFile) => {
  const composed = composeStories(stories);
  const testCases = Object.values(composed).map((Story: any) => [Story.storyName, Story, Story.args]);
  test.each(testCases)("renders %s", (_storyName, Component: any, args) => {
    if (_storyName.match(/Fake/)) return;

    const tree = render(<Component {...(args || {})} />);
    expect(tree.baseElement).toMatchSnapshot();
  });
};
