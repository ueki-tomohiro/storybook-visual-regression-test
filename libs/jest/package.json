{
  "name": "@demo-libs/jest",
  "version": "0.0.1",
  "private": true,
  "description": "Demo Jest",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.21.4",
    "@babel/preset-env": "^7.21.4",
    "@babel/preset-typescript": "^7.21.4",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.5.1",
    "@types/jest": "^29.5.1",
    "babel-jest": "^29.5.0",
    "babel-plugin-transform-import-meta": "^2.2.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0"
  },
  "engines": {
    "node": "^20",
    "yarn": ">=1"
  },
  "volta": {
    "node": "20.0.0",
    "yarn": "1.22.19"
  }
}
