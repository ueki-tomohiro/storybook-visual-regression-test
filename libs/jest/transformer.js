const babelJest = require("babel-jest");
const babelOptions = {
  presets: [
    [
      "@babel/env",
      {
        targets: {
          node: "current",
        },
        exclude: ["transform-regenerator"],
      },
    ],
    "@babel/preset-typescript",
    "@babel/react",
  ],
  plugins: ["babel-plugin-transform-import-meta"],
  babelrc: false,
};
module.exports = babelJest.createTransformer(babelOptions);
