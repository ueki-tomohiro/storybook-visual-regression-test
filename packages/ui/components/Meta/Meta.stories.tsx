import React from "react";

import { Meta, MetaProps } from "./Meta";

const defaultArgs: MetaProps = {
  pageTitle: "タイトル",
  children: <>Content</>,
};

export default {
  title: "components/Meta",
  component: Meta,
};

export const Default = { args: defaultArgs };
