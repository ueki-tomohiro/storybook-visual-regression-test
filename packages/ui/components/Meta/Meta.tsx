import Head from "next/head";
import React from "react";

export type MetaProps = React.PropsWithChildren<{
  pageTitle?: string;
  header?: React.ReactNode;
}>;

export const Meta: React.FC<MetaProps> = ({ pageTitle, header, children }) => {
  return (
    <>
      <Head>
        <title>{pageTitle ? `Demo - ${pageTitle}` : "Demo"}</title>
        {header}
      </Head>
      {children}
    </>
  );
};
