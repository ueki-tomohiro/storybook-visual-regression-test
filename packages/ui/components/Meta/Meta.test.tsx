import { renderFromStories } from "@demo-libs/jest";

import * as stories from "./Meta.stories";

describe(stories.default.title, () => {
  renderFromStories(stories);
});
