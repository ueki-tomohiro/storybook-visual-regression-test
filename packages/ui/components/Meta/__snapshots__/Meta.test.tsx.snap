// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Meta renders Default 1`] = `
<body>
  <div>
    Content
  </div>
</body>
`;
